### ISSUE TITLE
Accessibility: Order Received page missing H1 and skips heading levels

### ISSUE THREAD

**Issue by amberhinds** [woothemes/woocommerce#43624]
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

The order received page does not have the correct heading structure.
<img width="242" alt="Screenshot 2024-01-14 at 8 49 06 PM" src="https://github.com/woocommerce/woocommerce/assets/13153456/5a11ca9d-77bb-4e14-869b-ed77cd261fd2">

The expected H1 "Thank you. Your order has been received." is just large styled paragraph text. See:
https://github.com/woocommerce/woocommerce/blob/758df4854d7b627fc26eb5c0860dc492dea6a06e/plugins/woocommerce/templates/checkout/order-received.php#L23

The subsections on the page (Order Details, Shipping Address, Billing Address) use H3s when they should be H2s.

### Expected behavior

All pages should have an H1.

Headings on the page following the H1 should have the correct tag to create a meaningful outline of the page content.

### Actual behavior

There is no H1 or H2s on the page, only H3s.

### Steps to reproduce

Submit an order and inspect the heading levels on the page.

### WordPress Environment

N/A

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woo.com/products/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

**Comment by mikejolley** [comment]
The main paragraph at the top is a sentence describing the order status. For example, if the order was not paid due to an issue it would read:

```
Your order cannot be processed as the originating bank/merchant has declined your transaction. Please attempt your purchase again.
```

Therefore I don't think it would be suited as a heading element. We could either add a generic heading at template level, or add a new shorter heading (based on status) to the block itself. This will require some design input.

Merchants can customise this themselves in the meantime: Appearance > Templates > WooCommerce > Order Confirmation. You can insert headings above, or change heading levels of the inner blocks that are currently h3.

**Comment by mikejolley** [comment]
I looked into this some more to see what can be done for the order confirmation block.

**h3->h2**

This is a simple change to a few patterns. There were no visible issues in doing this.

**paragraph->h1** 

More complex because:

- The status block may contain a single sentence, or a few (for failed status).
- The copy is not a good fit for a heading
- Styling options only seem to apply to the paragraph

This is the current text for each status.

Default:

`Thank you. Your order has been received.`

Cancelled:

`Your order has been cancelled.`

Refunded:

`Your order was refunded %s.`

Completed:

`Thank you. Your order has been fulfilled.`

Failed:

`Your order cannot be processed as the originating bank/merchant has declined your transaction. Please attempt your purchase again.`

To use headings we could either include one before the current block content, or reword each to be more succinct e.g.`Thank you. Your order has been received.` -> `Order Received`

cc @nikkivias @ralucaStan

**Comment by amberhinds** [comment]
Likely adding a heading before would be best.

**Comment by ralucaStan** [comment]
> To use headings we could either include one before the current block content, or reword each to be more succinct e.g.Thank you. Your order has been received. -> Order Received

@mike I agree with @amberhinds. We can include a heading about the current content; This would be in sync with the classic experience. 

I don't think that we need a design for this. The theme should be able to style the H1 accordingly. 

<img width="856" alt="Screenshot 2024-06-14 at 14 52 36" src="https://github.com/woocommerce/woocommerce/assets/1628454/f14309da-fe66-4020-845c-630cc222378e">

Worth mentioning that merchants can remove this title, but at least they start from a good starting point by offering the recommended H1.

**Comment by mikejolley** [comment]
@ralucaStan Check the default styling of the order status block. It's styled like a heading, so it will look very strange with 2 elements styled like headings with similar content. In the legacy template the text was a regular paragraph.

![Screenshot 2024-06-14 at 14 14 03](https://github.com/woocommerce/woocommerce/assets/90977/6af587a0-f7ba-46f3-98c6-421b506706f6)

The heading was left out of the designs for the order confirmation template.

**Comment by Manussakis** [comment]
I have started working on this issue

**Comment by Manussakis** [comment]
After looking into this issue, these are my thoughts/questions:

1. I only see this being an issue for the Order confirmation page in Block themes (unless I missed something).
2. **h3 -> h2** – I have started working on it under this PR (WIP) https://github.com/woocommerce/woocommerce/pull/49317
3. **Adding h1 to the order confirmation page** – I see two options to address this:
    1. Leverage the `order-confirmation-status` block content and add the heading above the paragraph through a new filter, likely named `woocommerce_thankyou_order_received_title`. The con is that editors wouldn't be able to delete the title without deleting the paragraph since they are both in the same block.
    2. Create a new block called `order-confirmation-title` and place it before the `order-confirmation-status` on the order confirmation template.
    In both cases, the paragraph's style should be updated so it no longer looks like a heading.
4. What should the headings read for each one of the statuses? Does the following list make sense?
    - **Default** – Order received.
    - **Cancelled** – Order cancelled
    - **Refunded** – Order refunded
    - **Completed** – Order fulfilled
    - **Failed** – Order failed
  
cc: @jeffpaul

**Comment by jeffpaul** [comment]
@woocommerce/rubik checking to see if you all have preferences on approach based on @Manussakis' comment above?

**Comment by ralucaStan** [comment]
Thank you @Manussakis for presenting us with the 2 options. I am in favor of the 2nd option

> Create a new block called order-confirmation-title and place it before the order-confirmation-status on the order confirmation template. In both cases, the paragraph's style should be updated so it no longer looks like a heading.

Should we also enable editing the paragraph? Currently it's not editable in the editor. We can keep the title non-editable


> What should the headings read for each one of the statuses? Does the following list make sense?

From what I see [we only handle 1 status](https://github.com/woocommerce/woocommerce/blob/fa693249db3e89122dd5c9ef157cb8987364a0f2/plugins/woocommerce-blocks/assets/js/blocks/order-confirmation/status/edit.tsx). @mikejolley can confirm.

**Comment by mikejolley** [comment]
@ralucaStan 

> Should we also enable editing the paragraph? Currently it's not editable in the editor. We can keep the title non-editable

2 things to note:

1. The paragraph is filterable by plugins who want to customise it based on payment method or order status
2. The paragraph adapts to the order status e.g. on this template, a processing order could have different wording to completed or cancelled order.

I believe thats why we're asking about the various titles too; the text should adapt to order status. [I had a similar question last month.](https://github.com/woocommerce/woocommerce/issues/43624#issuecomment-2165816019)

We should ask @nikkivias to provide some guidance here to avoid overcomplicating things before we have proper direction.

**Comment by Manussakis** [comment]
I am following up on this issue to see if we have any guidance on how to address it.

cc: @nikkivias @mikejolley

**Comment by mikejolley** [comment]
Just recapping the issue so we can get feedback from @nikkivias.

This is what you see after submitting an order (order is processing) with the default template:

![Screenshot 2024-07-24 at 16 30 51](https://github.com/user-attachments/assets/6209cebf-3fee-42df-8ca8-b6ea1f7a3244)

The text "Thank you. Your order has been received." is not customisable because its generated based on the order status. This uses a paragraph tag.

If I make the order completed it looks like this:

![Screenshot 2024-07-24 at 16 31 34](https://github.com/user-attachments/assets/a7e71101-180a-4d05-8f09-31f6e7c86f6a)

And here is an example of a cancelled order:

![Screenshot 2024-07-24 at 16 31 54](https://github.com/user-attachments/assets/38fa0dcb-6076-411f-bf00-286a4881e168)

Some statuses inject extra content, for example, failed orders have some links to retry:

![Screenshot 2024-07-24 at 16 32 51](https://github.com/user-attachments/assets/1ae47f20-3a6d-4e4e-b57d-b170badb6b10)

**So the main questions we have are:**

1. Do we leave as as paragraph or convert to heading?
2. If we leave as a paragraph, should we inject a heading before? What should its content be, and how should both elements be styled?
3. If we convert to a heading, how does it get styled, and do we reword the default statuses?

cc @opr

**Comment by jeffpaul** [comment]
My two cents, but also curious what @amberhinds @amesplant @Manussakis  would recommend as well?

> 1. Do we leave as as paragraph or convert to heading?

I would leave as a paragraph.

> 2. If we leave as a paragraph, should we inject a heading before? What should its content be, and how should both elements be styled?

I would inject a heading before as "Order Status".

> 3. If we convert to a heading, how does it get styled, and do we reword the default statuses?

I would not convert the text to a heading, but rather leave as a paragraph and adding the heading as noted in 2 above.

**Comment by amesplant** [comment]
Here are my suggestions, albeit this is not based on what the code base is, but how I feel the page should behave, which pretty much mirror what @jeffpaul has said.

1. H1 which should be a generic title "Order Received" or whatever makes the most sense. I am not a big fan of leaving the H1 up to the content editor in as far as giving them the ability to **not** include it, it needs to be present.
2. The status is a paragraph block below the heading.
3. As far as the styles, stay consistent as much as possible across all pages with an H1. The status should be a style paragraph, and **not** appear to be a heading.

**Comment by mikejolley** [comment]
One option we were discussing was making the h1 present ~~for screenreaders~~ but not visible since it has little purpose, especially once its worded generically. Then the current structure could be maintained.

**Comment by Manussakis** [comment]
@mikejolley I don't believe having the H1 only visible for screen readers is the better approach. These are the reasons:
1. It breaks the concept of equivalent experience:

> An accessible design provides an equivalent user experience to all users, no matter how they may choose to interact with a product or service.

[Reference](https://jambysam.com/user-experience-ux-design-for-accessibility/#:~:text=Accessibility%20is%20the%20concept%20of,with%20a%20product%20or%20service.)

2. If merchants add a heading via Appearance > Templates > WooCommerce > Order Confirmation, the page will end up having two H1 elements, which is not ideal.

I am aligned with @jeffpaul and @amesplant suggestions.

1. Add a visible H1 with a generic text for all scenarios (Cancelled, Refunded, ...):  "Order Status"
2. The status is a paragraph block below the heading.
3. Update the status style to not appear to be a heading.

**Comment by mikejolley** [comment]
@Manussakis I mistyped—ignore the screenreader comment. This is not an accessibility issue, this issue is about page semantics. 

In this case it could be argued that a generic heading adds no value. The important content is the status of the order.

**Comment by amberhinds** [comment]
Hi,

I would agree with having a visible and short H1 for everyone. If it is easier to have a static rather than dynamic heading, then  "Order Status" seems to make sense for all the scenarios listed above.

The H1 doesn't have to explain the order status (canceled, success, failed, etc.) as long as it indicates that the order status can be found below it. Screen reader users would hear the heading and use it to jump to that section, then go below it for the relevant information.

I also agree that the paragraph should just be styled like a normal paragraph. Perhaps add classes and if a website owner wants to make it stand out differently they can do that in their theme.

**Comment by alexflorisca** [comment]
I noticed a related bug when testing for WC 9.2, thought I'd cross post it here in case you want to address it at the same time as this issue 

https://github.com/woocommerce/woocommerce/issues/50587

**Comment by Manussakis** [comment]
Thanks @alexflorisca ! I will assess the issue you mentioned once I start working on this one.

I am following up to confirm we came to a conclusion and I can start developing the solution.

The implementation would be:
1. Add a visible H1 with a generic text for all scenarios (Cancelled, Refunded, ...): "Order Status"
2. The status is a paragraph block below the heading.
3. Update the status style to not appear to be a heading.

**Comment by Manussakis** [comment]
@alexflorisca I assume your reaction with a thumbs-up to my last comment was about my response saying that I will assess  https://github.com/woocommerce/woocommerce/issues/50587 when I start working on this issue.

It seems we haven't concluded what needs to be implemented yet.

@nikkivias @opr  Do you have any thoughts to share on the implementations I listed in my last comment? 

I am tagging you since I noticed that @mikejolley did it before and I'd love to hear your thoughts.

**Comment by alexflorisca** [comment]
> @alexflorisca I assume your reaction with a thumbs-up to my last comment was about my response saying that I will assess https://github.com/woocommerce/woocommerce/issues/50587 when I start working on this issue.

Yes, thank you for looking at it. It's not a deal breaker if you can't include it as part of the fix for this issue.

**Comment by vikrampm1** [comment]
@opr @mikejolley can you review the comments above and let us know on how to best proceed forward here.

**Comment by mikejolley** [comment]
@vikrampm1 Thomas and co are at a meetup. I'll raise this in our next team meeting.

**Comment by jeffpaul** [comment]
@opr @mikejolley any chance y'all have had a chance to come to an agreement on approach here to unblock this issue for our team to work on a fix in Woo core?

**Comment by ralucaStan** [comment]
@jeffpaul we've discussed this and, as the label suggests, this ticket needs design input.

@elizaan36 would you be able to help out?

If not, to unblock this ticket I propose we only focus on adding a generic H1 ![Image](https://github.com/user-attachments/assets/9ce0adb2-eb31-4397-ae99-20eb9f457c30)

**Comment by ralucaStan** [comment]
Another option would be to wrap the top sentences, currently wrapped with `<p>`,  with an H1, keep the style and rephrase the texts:


`Thank you. Your order has been received.` -> `Your order was received`

`Your order has been cancelled.` -> `Your order has been cancelled`

`Your order was refunded %s.` -> `Your order has been refunded %s`

`Thank you. Your order has been fulfilled.` -> `Your order has been fulfilled`

`Your order cannot be processed as the originating bank/merchant has declined your transaction. Please attempt your purchase again.` -> `Your order cannot be processed as the bank/merchant declined the transaction. Please try again`

**Comment by elizaan36** [comment]
hey @mikejolley @ralucaStan @jeffpaul !

I think creating the H1 with "Order received" makes sense and as Raluca shared [in this comment](https://github.com/woocommerce/woocommerce/issues/43624#issuecomment-2168006780) the theme can style it. Regarding Mike's comment that the text string is styled as a heading, I think that's ok as long as the defaults play well together. Here's an example from when we redesigned the Order Confirmation template a while back. The H1 text default style can use a lighter font weight, which will help integrate and not distract from the main content.

![Image](https://github.com/user-attachments/assets/ec053ea5-2e23-4226-94f6-202d98c28ae5)
